src/shared/dns-resolver.h

```diff
@@ -0,0 +1,62 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <errno.h>
+#include "macro-fundamental.h"
```

Comment:
never include "macro-fundamental.h" directly, include "macro.h" instead, which transitively includes the fundamental version